### PR TITLE
Fix setup for development with docker

### PR DESCRIPTION
**Pull request by wvengen**
This completes #449. Reorganizes development and production docs somewhat, and brings back the development setup using docker. It could be improved, but I think this is already a small step forward.

After this is merged, the wiki page [Deployment notes](https://github.com/foodcoops/foodsoft/wiki/Deployment-notes) may need updated links.

### CODE REVIEW

**Comment by wvengen** [comment]
[This uid hack](https://chengl.com/using-docker-for-rails-development/) may still be useful, keeping that as a later improvement.

File: docker-compose-dev.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+version: '2'
+services:
+
+  web:
```
Comment (paroga) [inline docker-compose-dev.yml]: I'd like to align this with our foodcoops.net setup

File: docker-compose-dev.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    extends: worker
```
Comment (paroga) [inline docker-compose-dev.yml]: why does web extend worker and not the other way around?

File: docker-compose-dev.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    volumes:
+      - bundle:/home/app/bundle
+      - .:/home/app/src
+    environment:
+      - DATABASE_URL=mysql2://root:secret@mariadb/development?encoding=utf8
+      - REDIS_URL=redis://redis:6379
+      - QUEUE=foodsoft_notifier
+      - TEST_DATABASE_URL=mysql2://root:secret@mariadb/test?encoding=utf8
+
+  mailcatcher:
```
Comment (paroga) [inline docker-compose-dev.yml]: i see not configuration sending the mails to this container

File: docker-compose-dev.yml (hunk 1 above)
Comment (wvengen) [reply docker-compose-dev.yml]: Ok with that.

File: docker-compose-dev.yml (hunk 3 above)
Comment (wvengen) [reply docker-compose-dev.yml]: Took same approach as before, where current configuration detects docker environment variables and [connects to mailcatcher](https://github.com/foodcoops/foodsoft/blob/master/config/environments/development.rb.SAMPLE#L42). Later we can change this to a proper smtp url.

**Comment by wvengen** [comment]
Ok to merge? I'll do so today without response.

**Pull request merged**